handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -230,42 +230,44 @@ protected void deallocate() {
         this.jdkCompatibilityMode = jdkCompatibilityMode;
         Lock readerLock = context.ctxLock.readLock();
         readerLock.lock();
+        final long ssl;
         try {
-            ssl = SSL.newSSL(context.ctx, !context.isClient());
+            this.ssl = ssl = SSL.newSSL(context.ctx, !context.isClient());
```

Comment:
The suggestion in https://github.com/netty/netty/pull/6955#discussion_r126509924 was as follows:

```java
final long finalSsl;
Lock readerLock = context.ctxLock.readLock();
readerLock.lock();
try {
  finalSsl = SSL.newSSL(context.ctx, !context.isClient());
} finally {
  readerLock.unlock();
}

synchronized(this) {
  ssl = finalSsl;
...
}
```

Note that the assignment of `ssl` is delayed until inside the `synchronized` block ... so if other areas happen to read the value of `ssl` they *should* see it as "uninitialized" (e.g. `0`). However if we are not concerned about re-ordering we could just directly assign `this.ssl` outside the scope of the `synchronized` and rely on this to create the appropriate memory barrier for other locations (presumably using the same `synchronized` monitor) ... the first suggestion with the `final` variable was given because it seems more "straightforward".  However I'm not sure if there is value in having a `final long ssl` and also assigning `this.ssl` outside the `synchronized` block (please clarify if there is).